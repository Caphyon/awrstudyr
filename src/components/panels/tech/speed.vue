<template>
    <container>
      <template slot="header">
        <div class="section__header--stats">
          <speed-score :score="mobileScore" type="Mobile"></speed-score>
          <speed-score :score="desktopScore" type="Desktop"></speed-score>
        </div>
      </template>
      <div class="scrollable__container">
        <div>
          <h4 class="text--center text--muted mt0">Mobile</h4>
          <speed-item v-for="msg in mobileMessages" :data="msg" :key="msg.title"></speed-item>
        </div>
        <div>
          <h4 class="text--center text--muted mt0">Desktop</h4>
          <speed-item v-for="msg in desktopMessages" :data="msg" :key="msg.title"></speed-item>                
        </div>
      </div>
    </container>
</template>

<script>
import Vue from "vue";
import SpeedSection from "./../../sections/speed-section";
import Container from "./container";
import Item from "./items/speed-item";
Vue.component("container", Container);
Vue.component("speed-item", Item);
export default {
  mixins: [SpeedSection],
  data() {
    return {
      desc:
        "Page Speed Insights measures the performance of a page for mobile devices and desktop devices. It fetches the url twice, once with a mobile user-agent, and once with a desktop-user agent."
    };
  },
  methods: {
    extractMessages(type) {
      if (this.loading || !this.hasData) return [];
      const messages = this.data[type].formattedResults.ruleResults;
      const results = [];
      for (let key in messages) {
        const msg = messages[key];
        results.push({
          title: msg.localizedRuleName,
          level: msg.ruleImpact,
          msg: msg.summary
        });
      }
      return results;
    }
  },
  computed: {
    mobileMessages() {
      return this.extractMessages("mobile");
    },
    desktopMessages() {
      return this.extractMessages("desktop");
    }
  }
};
</script>