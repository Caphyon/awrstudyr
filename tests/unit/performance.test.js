import Vue from "vue";
import { shallowMount, config } from "@vue/test-utils";
import speed from "../../src/components/panels/tech/performance.vue"; // name of your Vue component
import speed_score from "../../src/components/panels/tech/items/performance-score.vue";
import speed_item from "../../src/components/panels/tech/items/performance-item.vue";


global.EventBus = new Vue();

describe("performance.vue", () => {

  config.methods["getPerformance"] = () => {};

  it("Testing speed component", () => {
    const wrapper = shallowMount(speed, {
      computed:{
        mobileMessages(){
          return {
            title: "First Contentful Paint",
            level: 0.63,
            msg: "First Contentful Paint marks the time at which the first text or image is painted. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint).",
          }
        },
        desktopMessages(){
          return {
            title: "First Contentful Paint",
            level: 0.63,
            msg: "First Contentful Paint marks the time at which the first text or image is painted. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint).",
          }
        }
      }
    });

    // controll if the component is instance
    expect(wrapper.isVueInstance).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('.section__header--stats.px2.py1').exists()).toBe(true);
    expect(wrapper.findAll(speed_score).wrappers.length).toBe(2);

    wrapper.find(speed_score).vm.$emit('score', 'type = "Mobile"');
    wrapper.find(speed_score).vm.$emit('score', 'type = "Desktop"');
    expect(wrapper.findAll('.scrollable__container').exists()).toBe(true);

    expect(wrapper.findAll('.text--center.text--muted.mt0.mb1').wrappers.length).toBe(2);
    wrapper.find(speed_item).vm.$emit('data', 'key');
    expect(wrapper.findAll(speed_item).wrappers.length).toBe(6);



  });

});
