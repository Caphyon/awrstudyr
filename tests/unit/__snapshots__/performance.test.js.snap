// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performance.vue Testing speed component 1`] = `
<container-stub>
  <template>
    <div
      class="section__header--stats px2 py1"
    >
      <performance-score-stub
        score="0"
        type="Mobile"
      />
       
      <performance-score-stub
        score="0"
        type="Desktop"
      />
    </div>
  </template>
   
  <div
    class="scrollable__container"
  >
    <div>
      <h4
        class="text--center text--muted mt0 mb1"
      >
        Mobile
      </h4>
       
      <performance-item-stub
        data="First Contentful Paint"
      />
      <performance-item-stub
        data="0.63"
      />
      <performance-item-stub
        data="First Contentful Paint marks the time at which the first text or image is painted. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint)."
      />
    </div>
     
    <div>
      <h4
        class="text--center text--muted mt0 mb1"
      >
        Desktop
      </h4>
       
      <performance-item-stub
        data="First Contentful Paint"
      />
      <performance-item-stub
        data="0.63"
      />
      <performance-item-stub
        data="First Contentful Paint marks the time at which the first text or image is painted. [Learn more](https://developers.google.com/web/tools/lighthouse/audits/first-contentful-paint)."
      />
    </div>
  </div>
   
  <!---->
</container-stub>
`;
